File: JJOOwebServiceAndroidClient/app/src/main/java/com/qindel/jjoowebserviceandroidclient/OlimpiadaAdapter.java
package com.qindel.jjoowebserviceandroidclient;

import android.content.Context;
import android.support.annotation.IdRes;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.BaseAdapter;
import android.widget.ListAdapter;
import android.widget.TextView;

import java.sql.Array;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Created by Qindel on 04/08/2017.
 */

public class OlimpiadaAdapter extends BaseAdapter {

    private ArrayList<HashMap<String,String>> data;

    public OlimpiadaAdapter(ArrayList<HashMap<String,String>> data){
        this.data = data;
    }

    @Override
    public int getCount() {
        return data.size();
    }

    @Override
    public Object getItem(int i) {
        return data.get(i);
    }

    @Override
    public long getItemId(int i) {
        return data.get(i).hashCode();
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        if(view == null){
            LayoutInflater inflater = (LayoutInflater) viewGroup.getContext()
                    .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            view = inflater.inflate(R.layout.main_list_item, viewGroup, false);
        }

        ((TextView) view.findViewById(R.id.id_ciudad)).setText(data.get(i).get("ID_CIUDAD"));
        ((TextView) view.findViewById(R.id.id_pais)).setText(data.get(i).get("ID_PAIS"));
        ((TextView) view.findViewById(R.id.nombre_pais)).setText(data.get(i).get("NOMBRE_PAIS"));
        ((TextView) view.findViewById(R.id.nombre_ciudad)).setText(data.get(i).get("NOMBRE_CIUDAD"));
        ((TextView) view.findViewById(R.id.valor)).setText(data.get(i).get("VALOR"));
        ((TextView) view.findViewById(R.id.veces_sede)).setText(data.get(i).get("NUMERO_VECES_SEDE"));
        ((TextView) view.findViewById(R.id.tipo_jjoo)).setText(data.get(i).get("DESCRIPCION_TIPO"));

        return view;
    }
}

File: JJOOxamarin/JJOOxamarin/JJOOxamarin.Android/views/OlimpiadaAdapter.cs
using Android.Widget;
using JJOOxamarin.model;
using System.Collections.Generic;
using Android.Views;
using System;
using JJOOxamarin.Droid;
using Android.App;

namespace JJOOxamarin.Droid.views
{
    class OlimpiadasAdapter : BaseAdapter<Olimpiada>
    {
        public List<Olimpiada> Lista { get; private set; }
        public Activity Context { get; private set; }
        public int LayoutID { get; private set; }

        public OlimpiadasAdapter(Activity context, List<Olimpiada> lista)
        {
            this.Lista = lista;
            this.Context = context;
            this.LayoutID = LayoutID;
        }

        public override Olimpiada this[int position] => this.Lista[position];

        public override int Count => this.Lista.Count;

        public override long GetItemId(int position)
        {
            return this.Lista[position].GetHashCode();
        }

        public override View GetView(int position, View convertView, ViewGroup parent)
        {
            var olimpiada = this[position];
            View view = convertView;
            if (view == null)
            {
                view = Context.LayoutInflater.Inflate(Resource.Layout.OlimpiadaView, null);
            }
            view.FindViewById<TextView>(Resource.Id.id_ciudad).Text = olimpiada.id_ciudad.ToString();
            view.FindViewById<TextView>(Resource.Id.id_pais).Text = olimpiada.id_pais.ToString();
            view.FindViewById<TextView>(Resource.Id.nombre_ciudad).Text = olimpiada.nombre_ciudad;
            view.FindViewById<TextView>(Resource.Id.nombre_pais).Text = olimpiada.nombre_pais;
            view.FindViewById<TextView>(Resource.Id.valor).Text = olimpiada.valor.ToString();
            view.FindViewById<TextView>(Resource.Id.veces_sede).Text = olimpiada.numero_veces_sede.ToString();
            view.FindViewById<TextView>(Resource.Id.tipo_jjoo).Text = olimpiada.descripcion_tipo_jjoo;
            return view;
        }
    }
}
